### PR TITLE
Implement IF THEN statement

### PR DESCRIPTION
**Pull request by gferon**
TODO:
- [x] Check if naming of structs and enums if consistent with spec.
- [x] Find out why the diff of the test output does not pass

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -71,6 +72,9 @@ impl fmt::Display for Error {
             ),
             Error::UnexpectedReturn { src_line_number } => {
                 write!(f, "{}: error: unexpected return\n", src_line_number)
+            },
+            Error::InvalidIfStatement => {
+                write!(f, "invalid IF statement, it only supports string-string or numeric-numeric comparisons\n")
```
Comment (boxdot) [inline src/error.rs]: I think we need to add `error: ` prefix here, to be conform with other messages.

File: src/parser.rs [hunk 2]
```diff
@@ -229,6 +243,24 @@ named!(gosub_statement<CompleteStr, Statement>,
         (Statement::Gosub(line_number))
     ));
 
+// TODO: Restrict to numericexpr-relation-numericexpr
+// or stringexpr-relation-stringexpr
+named!(if_then_statement<CompleteStr, Statement>,
+    do_parse!(
```
Comment (boxdot) [inline src/parser.rs]: I think you can do an alt! tag here, and use `string_expression` or `numeric_expression` to restrict resolve your TODO.

**Comment by boxdot** [comment]
Don't forget to update the progress in README.

File: src/parser.rs (hunk 2 above)
Comment (gferon) [reply src/parser.rs]: I fail on runtime, which adheres slightly better to the nature of BASIC eheh

Comment (boxdot) [reply src/parser.rs]: But then this is not according to the spec, which explicetly sais that this should not be possible.

Comment (gferon) [reply src/parser.rs]: Does the spec separate the parsing and interpreting part?

Comment (boxdot) [reply src/parser.rs]: It says: `relational-expression = numeric-expression relation numeric-expression | string-expression relation string-expression`. Right now, we have `relational-expression = expression relation expression`.

File: src/error.rs [hunk 3]
```diff
@@ -28,6 +28,9 @@ pub enum Error {
     UnexpectedReturn {
         src_line_number: u16,
     },
+    InvalidIfStatement {
+        line_number: u16,
```
Comment (jeschkies) [inline src/error.rs]: I think this should be `src_line_number`.

File: src/interpreter.rs [hunk 4]
```diff
@@ -212,6 +261,19 @@ where
...
         Statement::Gosub(line_number) => Action::Gosub(*line_number),
+        Statement::If(left_expression, relationship, right_expression, line_number) => {
+            if evaluate_if(
+                left_expression,
+                relationship,
+                right_expression,
+                line_number,
+                state,
+            )? {
+                Action::Goto(*line_number)
```
Comment (jeschkies) [inline src/interpreter.rs]: What do you think about return the action from `evaluate_if`?

Comment (gferon) [reply src/interpreter.rs]: then it's not exactly consistent with the other evaluate_* calls is it? I like seeing what `Action` are taken in the core loop of the state machine.

Comment (jeschkies) [reply src/interpreter.rs]: Ah. True. I thought we always returned the `Action`.

File: src/ast.rs [hunk 5]
```diff
@@ -284,6 +290,24 @@ pub enum LetStatement {
...
+// 12. Control statements
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Relation {
+    LessThan,
+    LessThanOrEqualTo,
+    EqualTo,
+    GreaterThan,
+    GreaterThanOrEqualTo,
+    NotEqualTo,
```
Comment (jeschkies) [inline src/ast.rs]: How is this related to `EqualityRelation::NotEqualTo`?

Comment (gferon) [reply src/ast.rs]: @jeschkies it's like this in the spec, I know it sounds weird. I'll double check.
